File: C++/409-Longest_Palindrome.cpp
// Runtime: 12 ms, faster than 5.37% of C++ online submissions for Longest Palindrome.
// Memory Usage: 10.5 MB, less than 6.67% of C++ online submissions for Longest Palindrome.
class Solution {
public:
    int longestPalindrome(string s) {
        set<int> se;
        for (char c : s)
            if (se.find(c) == se.end())
                se.insert(c);
            else
                se.erase(c);
        return se.size() > 0 ? s.size() - se.size() + 1 : s.size();
    }
};

File: Algorithms/Python3.x/409-Longest_Palindrome.py
# Runtime: 36 ms, faster than 84.94% of Python3 online submissions for Longest Palindrome.
# Memory Usage: 13.9 MB, less than 8.33% of Python3 online submissions for Longest Palindrome.
class Solution:
    def longestPalindrome(self, s: str) -> int:
        se = set()
        for c in s:
            if c not in se:
                se.add(c)
            else:
                se.remove(c)
        return len(s) - len(se) + 1 if len(se) > 0 else len(s)  # len(hash) is the number of the odd letters
